### DISCUSSION TITLE
Drag a control point in a Plot

### DISCUSSION THREAD

**Discussion by huand** [emilk/egui#3082] [Q&A]
Is it possible to drag a control point in a Plot? Similarly to the Bezier Demo but doing it in a plot instead of a canva? for example moving the extremities of a line? There is no "allocate_painter" for plots to get an early response. Did someone managed to do something similar to this question already?
thanks!

**Comment by haricot** [comment]
In the idea I proposed to recover the indices of the PreparedPlot::hover function via this PR https://github.com/emilk/egui/pull/2628.
I'm refactoring it to provide a clearer solution, but I've been pretty busy lately.
it works and it can be customized if you allow to pass a specific hover function in the initialization options of the plot.
I think it's a valid solution otherwise I'm also interested in knowing the alternative "picking" or "selection by pixel" method.
The best of both worlds would therefore probably be to use this quick approach first, and switch to traditional picking if too many objects are present in the tested neighborhood, to ensure accurate object selection.
